controller/linodemachine_controller.go

```diff
@@ -370,31 +297,12 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 		return nil, err
 	}
 
-	if linodeInstance == nil {
-		err = errors.New("missing instance")
-
-		logger.Error(err, "Panic! Failed to create isntance")
-
-		return nil, err
-	}
```

Comment:
I'm not totally sure that we can remove this check? I assume it double-checks this part: https://github.com/linode/cluster-api-provider-linode/blob/9a5dfbdfd473c8e8b8b1cd38ba72a38cc9232ea4/controller/linodemachine_controller.go#L246-L251

But I suspect this might be related to a similar issue with the Linode Firewall, where if you call the `linodego.Client` with an incorrect `util.Filter` you can get a `nil` object back with no error.